SparkPost/matchbox
# FE-810 Update ActionList

### What Changed
- Restyles ActionList
- Removes ActionList `groupByKey` default prop
- Adds Action proptypes, used internally by ActionList
- ActionList actions now accept a new `is` prop, and accept either `link` `button` or
  `checkbox`

<img width="360" alt="Screen Shot 2020-03-23 at 1 39 14 PM" src="https://user-images.githubusercontent.com/3903325/77345940-bbbcc580-6d0b-11ea-8f10-cf34d1c0d925.png">

### How To Test or Verify
- Run storybook with `npm run start:storybook`
- Verify ActionList stories

### PR Checklist
